geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync } from 'fs';
+const mapshaper = require('mapshaper');
+const unzipper = require('unzipper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private _unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          unzipper.Extract({
+            path: fileInfo.destination + fileInfo.filename.replace('.zip', ''),
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted succesfully`),
+        )
+        .on('error', () =>
+          reject(new Error(`${fileInfo.filename} could not been extracted`)),
+        );
+    });
+  }
+
+  private async _shapeFileToGeoJson(fileInfo: Express.Multer.File) {
+    const outputKey = `shapefile-${new Date().getTime()}.gejson`;
+    const _geoJson = await mapshaper.applyCommands(
+      `-i ${fileInfo.path.replace('.zip', '')}/*.shp -o ${outputKey}`,
```

Comment:
Would be great to provide unit tests on sample file. I believe it does not fully parse shapefile (coordinates are missing)